src/js/components/Select/Select.js

```diff
@@ -267,11 +267,27 @@ const Select = forwardRef(
 
     // element to show, trumps inputValue
     const selectValue = useMemo(() => {
-      if (valueLabel) return valueLabel;
-      if (React.isValidElement(value)) return value; // deprecated
+      if (valueLabel instanceof Function) {
+        if (value) return valueLabel(value);
+      } else if (valueLabel) return valueLabel;
```

Comment:
If valueLabel is a function but there is no value it will return undefined. When selectValue is undefined the normal SelectTextInput is displayed (line 406)